lazy_load_symbols/load_sym_lazily.py

```diff
@@ -0,0 +1,171 @@
+    A command to load a recording without loading
+    the symbols and then loading only the required symbols.
+    '''
+
+    def __init__(self):
+        super(LoadLazySymbols, self).__init__('loadlazy', gdb.COMPLETE_EXPRESSION)
+        self.solist = []
+        self.sodict = {}
+
+    def load_so_list(self):
+        self.solist = gdb.execute('info sharedlibrary', to_string=True)
+        solibre = r'0x([0-9a-f]+)\s+0x([0-9a-f]+).+ ([^\0]+)\/(.+)'
+        solibc = re.compile(solibre)
+        for so in self.solist.split('\n'):
+            m = solibc.match(so)
+            if m:
+                name = m.group(4)
+                so_path = m.group(3)
+                sa = m.group(1)
+                ea = m.group(2)
+                if name not in self.sodict:
+                    self.sodict[name] = {}
+                    self.sodict[name]['symbols'] = {}
+                    self.sodict[name]['fullname'] = so_path + '/' + name
+                    self.sodict[name]['start'] = sa
+                    self.sodict[name]['end'] = ea
+                    self.sodict[name]['loaded'] = False
+                else:
+                    # this shouldn't happen but we can just verify it is indeed the same library
+                    assert self.sodict[name]['start'] == sa, 'found same' \
+                        ' library ({}) at 2 different locations: {} != {}' \
+                        .format(name, sa, self.sodict[name]['start'])
+                    assert self.sodict[name]['end'] == ea, 'found same library' \
+                        ' ({}) ending at 2 different locations: {} != {}' \
+                        .format(name, ea, self.sodict[name]['end'])
+        return self.sodict
+
+    def load_all_symbols(self):
+        for so in self.sodict:
+            fn = self.sodict[so]['fullname']
+            nm = Popen(['nm', '-D', fn], stdout=PIPE)
+            (symlist, err) = nm.communicate()
+            assert not err, 'nm failed somehow {}'.format(err)
+            assert symlist, 'no symbols in file {}'.format(fn)
+            ec = nm.wait()
+            assert not ec, 'nm exited with {}'.format(ec)
+            self.sodict[so]['symbols'] = symlist
+
+    def invoke(self, rec, from_tty):
+        if self.sodict:
+            pass
+        else:
+            gdb.execute('set auto-solib-add off', to_string=True)
+            gdb.execute('uload {}'.format(rec))
+
+
+class PopulateBT(gdb.Command):
+    '''
+    A command to read the symbols for an empty backtrace
+    '''
+
+    def __init__(self):
+        super(PopulateBT, self).__init__('popbt', gdb.COMPLETE_EXPRESSION)
+        self.sodict = {}
+
+    def get_frame_name(self, pc):
+        for name in self.sodict:
+            sa = int(self.sodict[name]['start'], 16)
+            ea = int(self.sodict[name]['end'], 16)
+            if pc >= sa and pc < ea:
+                gdb.execute('sharedlibrary {}'.format(name))
+
+    def invoke(self, arg, from_tty):
+        ll = LoadLazySymbols()
```

Comment:
This is going to create a new command which overrides the old command.
Instead of doing this, why not having `load_so_list` as a standalone function used both by `LoadLazySymbols` and `PopulateBT`?

Reply:
indeed, it is better